### PR TITLE
Add Admin Role Guard and Implement User Fetching Endpoints

### PR DESCRIPTION
**Pull request by saralopezbz**
#### Features
- **Admin Role Guard**
  - Added a guard (`RolesGuard`) to restrict access to specific views to users with the Admin role.
  - Applied the guard to all routes that require administrative access.

- **User Fetching Endpoints**
  - Implemented various endpoints for fetching user data based on different criteria:
    - **GET `/users`**: Retrieve all registered users.
      - **User Story**: As an administrator, I want to see all registered users to know their details.
    - **GET `/users/country`**: Retrieve users by country.
      - **User Story**: As an administrator, I want to see all users from a specific country to understand where we have the most users.
    - **GET `/users/query`**: Retrieve users by name and email.
      - **User Story**: As an administrator, I want to search for all registered users by name or email to know their details.
    
  - Secured the endpoints using the `RolesGuard` to ensure only administrators can access them.

#### Implementation Details
- **Admin Role Guard**:
  - The `RolesGuard` has been implemented and integrated into the user fetching endpoints to restrict access to users with the Admin role.
- **Endpoints**:

  - **`@Get('/users')`**: Fetch all users (Admin only).
  - **`@Get('/users/country')`**: Fetch users by country (Admin only).
  - **`@Get('/users/query')`**: Fetch users by name and email (Admin only).

#### Testing
- Verified that only users with the Admin role can access protected views.
- Tested user fetching endpoints to ensure they return the correct data and handle edge cases appropriately.

#### Testing Endpoints
- **GET** `http://localhost:3000/users/`: Obtain all users (Admin only)
- **GET** `http://localhost:3000/users/query?email=testing@example.com`: Obtain users by email (Admin only)
- **GET** `http://localhost:3000/users/country?country=Chile`: Obtain users by country (Admin only)

### CODE REVIEW

File: src/auth/guards/roles.guard.ts
Comment (sara-rioseco) [inline src/auth/guards/roles.guard.ts:1]: I think you implemented both auth and roles guard in the same file, maybe we could separate this to keep our code cleaner. Actually the guards were already created, they only had to be implemented inside the controller. Changing this files was not necessary.

File: src/users/users.controller.ts
Comment (sara-rioseco) [inline src/users/users.controller.ts:1]: I think for the findAll/findByCountry/findByQuery methods we could use a single endpoint, with optional query parameters. Also, if we leave it like this, you will be constructing the url as /users/country?country=chile which doesn't look so good. We can talk about this tomorrow!

File: src/users/users.service.ts
Comment (sara-rioseco) [inline src/users/users.service.ts:1]: findOne method is perfect. But the same thing that happened in the controller happens here. We don't really need a findByCountry or findByQuery method if we add everything as optional query parameters inside the findAll method. It's just a suggestion, so we better wait to see what @LucianoFromTrelew thinks.

Comment (LuchoSerruya) [reply src/users/users.service.ts:1]: Yeah I totally agree with Sara - you can expand the `findByQuery` method to receive either a name, an email, or a country and based on received params, create the query to execute

**Review by LuchoSerruya — CHANGES_REQUESTED** [review]
Pending changes

**Review by sara-rioseco — COMMENTED** [review]
Changes are done!

**Review by sara-rioseco — APPROVED** [review]

**Review by LuchoSerruya — APPROVED** [review]

**Pull request merged**